functions/common/datastore.js

```diff
@@ -66,12 +66,37 @@ class Datastore {
 
   fetchForm(projectId, featureTypeId, formId) {
     return this.fetchDoc_(
-        `projects/${projectId}/featureTypes/${featureTypeId}/forms/${formId}`);
+      `projects/${projectId}/featureTypes/${featureTypeId}/forms/${formId}`);
   }
 
   fetchSheetsConfig(projectId) {
     return this.fetchDoc_(`projects/${projectId}/sheets/config`);
   }
+
+  insertFeature(projectId, layerId, { caption, location }) {
+    var docRef = this.db_.collection('projects').doc(projectId);
+    return docRef.get().then((docSnapshot) => {
```

Comment:
I think if the project doesn't exist, we'd want to fail rather than creating a new project; the project should have already been created via the web UI. I believe we get this for free when calling `.add()`; it will throw an exception if the ancestor collections don't exist.